### PR TITLE
Add cli retry command to requeue event deliveries

### LINKED ISSUE #406: Add a requeue mechanism to pick up event deliveries that are stuck in the scheduled state
So essentially, convoy should provide a cli interface to force the sending of jobs that are stuck in the scheduled state in the queue. Like so:
```bash
convoy retry --time=1m --status=scheduled
```

so read from db, and check if they're in taskq, if not write it back, or just delete from the queue and write it back.

### PR DESCRIPTION
**Pull request by danvixent**
fixes #406

### CODE REVIEW

**Comment by jirevwe** [comment]
@danvixent is this ready?

**Comment by danvixent** [comment]
@jirevwe no, currently testing.

@oneEyedSunday RunE returns an error, Run does not

**Comment by oneEyedSunday** [comment]
Thanks

**Review by ogbanugot — APPROVED** [review]

File: cmd/retry.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+			default:
+				log.WithError(err).Fatalf("the retry command only works with redis queue for now")
+			}
+
+			var wg sync.WaitGroup
+
+			wg.Add(1)
+			go processEventDeliveryBatches(ctx, a, deliveryChan, q, &wg)
+
+			for {
```
[the remark this answers is not in the document]
Comment (danvixent) [reply cmd/retry.go:66]: for it to be effectively empty, we expect an error e.g ErrNoDocuments

File: cmd/retry.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+		Short: "retry event deliveries with a particular status in a timeframe",
+		Run: func(cmd *cobra.Command, args []string) {
+
+			d, err := time.ParseDuration(timeInterval)
+			if err != nil {
+				log.WithError(err).Fatal("failed to parse time duration")
+			}
+
+			s := datastore.EventDeliveryStatus(status)
+			if !s.IsValid() {
```
[the remark this answers is not in the document]
Comment (danvixent) [reply cmd/retry.go:31]: I think this gives us the "replay" functionality.
cc @Subomi @jirevwe @ogbanugot

**Review by Dotunj — APPROVED** [review]

**Pull request merged**